houston/deployment.go

```diff
@@ -551,7 +573,7 @@ func (h ClientImplementation) GetDeployment(deploymentID string) (*Deployment, e
 		return nil, handleAPIErr(err)
 	}
 
-	return &res.Data.GetDeployment, nil
+	return res.Data.GetDeployment, nil
```

Comment:
nit: Do we expect more than one deployment to be returned by Houston for a query by id? If not, then I would say it might be simpler to check the length of the response and return just the first element, that way function signature would remain the same and we won't require any further changes in the caller functions.

Or if we go by what's currently implemented then in all caller functions instead of directly fetching the first element, we would still need to add check on the length of the array first.

```Golang
if len(res.Data.GetDeployment) > 0 {
        return &res.Data.GetDeployment[0], nil
}
return nil, fmt.Errorf("deployment with id %s not found", deploymentID)
```